lib/exceptional_synchrony/callback_exceptions.rb

```diff
@@ -38,6 +38,40 @@ def return_exception(*args)
           ex
         end
       end
+
+      private
+
+      def handle_result_error(result)
+        error = result.error
+        if error_is_a_timeout?(error)
+          raise Timeout::Error
+        elsif error_is_a_syscall_class?(error)
+          raise Failure, format_failure(error.to_s, format_inspect(result))
+        else
+          raise Failure, format_failure(format_inspect(error), format_inspect(result))
+        end
+      end
+
+      def format_failure(error_string, result_string)
```

Comment:
As we discussed in person, how about changing this contract to pass an optional argument for the error. That way it can be encapsulated in one place how to format that. In fact, you could make that method `raise` the `Failure` exception too.